holmes/core/tool_calling_llm.py

```diff
@@ -474,29 +499,66 @@ def call(  # type: ignore
                 for future in concurrent.futures.as_completed(futures):
                     tool_call_result: ToolCallResult = future.result()
 
-                    tool_number = (
-                        futures_tool_numbers[future]
-                        if future in futures_tool_numbers
-                        else None
-                    )
-
-                    if (
-                        tool_call_result.result.status
-                        == StructuredToolResultStatus.APPROVAL_REQUIRED
-                    ):
-                        tool_call_result = self._handle_tool_call_approval(
-                            tool_call_result=tool_call_result,
-                            tool_number=tool_number,
-                            trace_span=trace_span,
-                        )
-
-                    tool_result_response_dict = (
-                        tool_call_result.as_tool_result_response()
-                    )
-                    tool_calls.append(tool_result_response_dict)
-                    all_tool_calls.append(tool_result_response_dict)
+                    tool_calls.append(tool_call_result.as_tool_result_response())
                     messages.append(tool_call_result.as_tool_call_message())
-                    tokens = self.llm.count_tokens(messages=messages, tools=tools)
+
+                    # 🔍 Debug: Log every tool result for termination analysis
+                    logging.info(f"🔍 TERMINATION DEBUG: Tool {tool_call_result.tool_name} status={tool_call_result.result.status}")
```

Comment:
If these code are specific to kaito, let's move them to a function to make the main default logic clean.